crates/proof-of-sql/src/sql/proof/proof_builder.rs

```diff
@@ -99,16 +99,19 @@ impl<'a, S: Scalar> ProofBuilder<'a, S> {
         level = "debug",
         skip_all
     )]
-    pub fn commit_intermediate_mles<V: VecCommitmentExt>(
+    pub fn commit_intermediate_mles<C: Commitment>(
         &self,
         offset_generators: usize,
-        setup: &V::CommitmentPublicSetup<'_>,
-    ) -> V {
-        V::from_commitable_columns_with_offset(
+        setup: &C::PublicSetup<'_>,
+    ) -> Vec<C> {
+        let mut commitments = vec![C::default(); self.commitment_descriptor.len()];
+        C::compute_commitments(
+            &mut commitments,
```

Comment:
NIT: is it possible to compute the commitments into a `Vec<C>` directly if you import `VecCommitmentExt`?

```suggestion
        let commitments = Vec::<C>::from_committable_columns_with_offset(
```